platforms/ide/tooling-api/src/crossVersionTest/groovy/org/gradle/integtests/tooling/r33/CacheableTaskOutcomeCrossVersionSpec.groovy

```diff
@@ -73,24 +71,6 @@ class CacheableTaskOutcomeCrossVersionSpec extends ToolingApiSpecification {
         cacheableTaskResult(pullFromCacheResults).upToDate
     }
 
-    @ToolingApiVersion('<3.3 >=3.0')
-    @TargetGradleVersion('>=3.5')
```

Comment:
❓Shouldn't this be in the accepted range?

Reply:
Especially `@TargetGradleVersion('>=3.5')`, as it's higher than `3.0`

Reply:
Regarding your main comments, I don't want to duplicate information, but mentioning the rules in the docs would not hurt. I'll add something.

Reply:
At least point to the document

Reply:
> We should only test TAPI 7+ versions.
But my main concern is about the Gradle version. Isn't the by-default `@TargetGradleVersion('>=3.0')`? Wouldn't that mean that we push the lower boundary of the Gradle version lower?

Reply:
If I understand your question correctly, you are asking whether the annotations combine or overwrite each other. It's the latter. If the base class has `@TargetGradleVersion('>=4.0 <=6.0')` and a test method has `@TargetGradleVersion('>=5.0 <=7.0')`, the method will be execute with Gradle versions 5.0, 6.0, 7.0 (and with the minor releases in between, of course).

Reply:
Added some documentation, ptal.

Reply:
> you are asking whether the annotations combine or overwrite each other

here we go, probably this is why I didn't understand the situation!